services/pull/pull.go

```diff
@@ -838,3 +866,158 @@ func IsHeadEqualWithBranch(ctx context.Context, pr *issues_model.PullRequest, br
 	}
 	return baseCommit.HasPreviousCommit(headCommit.ID)
 }
+
+// GetCodeOwners returns the code owners configuration
+// Return empty slice if files missing
+// Return error on file system errors
+// We're trying to do the best we can when parsing a file.
+// Invalid lines are skipped. Non-existent users and groups too.
+func GetCodeOwners(ctx context.Context, repo *repo_model.Repository, branch string) ([]*CodeOwnerRule, error) {
+	files := []string{"CODEOWNERS", "docs/CODEOWNERS", ".gitea/CODEOWNERS"}
+	gitRepo, closer, err := git.RepositoryFromContextOrOpen(ctx, repo.RepoPath())
+	if err != nil {
+		return nil, err
+	}
+	defer closer.Close()
+
+	if !gitRepo.IsBranchExist(branch) {
+		return nil, &git.ErrBranchNotExist{Name: branch}
+	}
+
+	var data []byte
+	for _, file := range files {
+		data, err = gitRepo.GetFileContent(branch, file)
+		if err == nil {
+			break
+		}
+	}
+
+	if len(data) == 0 {
+		return nil, nil
+	}
+
+	rules := make([]*CodeOwnerRule, 0)
+	lines := strings.Split(string(data), "\n")
+
+	for _, line := range lines {
+		tokens := tokenizeCodeOwnersLine(line)
+		if len(tokens) == 0 {
+			continue
+		} else if len(tokens) < 2 {
+			log.Info("Incorrect codeowner line: %s", line)
```

Comment:
```suggestion
```

I don't think log at here is necessary

Reply:
Without logs, it's hard to debug why the line is being silently ignored.

Reply:
looks you want user know whether the `codeowner` file has wrong format, isn't it? but this log is a report for server manager, not user. I think you can add a format check in file view like this : https://github.com/go-gitea/gitea/pull/24267

Reply:
Thanks for good suggestion. Added